nx/source/services/set.c

```diff
@@ -346,3 +346,38 @@ Result setsysGetColorSetId(ColorSetId* out)
     return rc;
 
 }
+
+Result setsysGetSerialNumber(char *serial) {
+    IpcCommand c;
+    ipcInitialize(&c);
+
```

Comment:
Add: if (serial) memset(serial, 0, 0x19);

Reply:
Should I change the declaration as well as the memcpy to 0x19 too?

Reply:
This is to make sure the string is NUL-terminated.

Reply:
Yes I was aware of that, but I guess the result buffer should be fine then.